sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,170 @@
+        :param partition_processor_callable: a callable that is called to return a PartitionProcessor
+        :param partition_manager: an instance of a PartitionManager implementation
+        :param initial_event_position: the offset to start a partition consumer if the partition has no checkpoint yet
+        """
+        self._consumer_group_name = consumer_group_name
+        self._eventhub_client = eventhub_client
+        self._eventhub_name = eventhub_client.eh_name
+        self._partition_processor_factory = partition_processor_factory
+        self._partition_manager = partition_manager
+        self._initial_event_position = kwargs.get("initial_event_position", "-1")
+        self._max_batch_size = eventhub_client.config.max_batch_size
+        self._receive_timeout = eventhub_client.config.receive_timeout
+        self._tasks: List[asyncio.Task] = []
+        self._instance_id = str(uuid.uuid4())
+        self._partition_ids = None
+
+    async def start(self):
+        """Start the EventProcessor.
+            1. retrieve the partition ids from eventhubs
+            2. claim partition ownership of these partitions.
+            3. repeatedly call EvenHubConsumer.receive() to retrieve events and
+            call user defined PartitionProcessor.process_events()
+        """
+        log.info("EventProcessor %r is being started", self._instance_id)
+        partition_ids = await self._eventhub_client.get_partition_ids()
+        self.partition_ids = partition_ids
+        claimed_list = await self._claim_partitions()
+        await self._start_claimed_partitions(claimed_list)
+
+    async def stop(self):
+        """Stop all the partition consumer
+
+        It sends out a cancellation token to stop all partitions' EventHubConsumer will stop receiving events.
+
+        """
+        for task in self._tasks:
+            task.cancel()
+            # It's not agreed whether a partition manager has method close().
+        log.info("EventProcessor %r has been cancelled", self._instance_id)
+
+    async def _claim_partitions(self):
+        partitions_ownership = await self._partition_manager.list_ownership(self._eventhub_name, self._consumer_group_name)
+        partitions_ownership_dict = dict()
+        for ownership in partitions_ownership:
+            partitions_ownership_dict[ownership["partition_id"]] = ownership
+
+        to_claim_list = []
+        for pid in self.partition_ids:
+            p_ownership = partitions_ownership_dict.get(pid)
+            if p_ownership:
+                to_claim_list.append(p_ownership)
+            else:
+                new_ownership = dict()
+                new_ownership["eventhub_name"] = self._eventhub_name
+                new_ownership["consumer_group_name"] = self._consumer_group_name
+                new_ownership["instance_id"] = self._instance_id
+                new_ownership["partition_id"] = pid
+                new_ownership["owner_level"] = 1  # will increment in preview 3
```

Comment:
suggest module "const" for `OWNER_LEVEL` instead of inline magic number. Also, maybe personal pref but I'd prefer to see one statement to construct the dict, either passing the `dict` initializer keywords args, or with a literal dict `{....}`

Reply:
Yes I see a lot of that Python code. 
This is more of Java style